### PR TITLE
Feature/app compatibility with environment

### PR DESCRIPTION
**Pull request by csadorf**
This is a large update to the home app to support the features introduced in https://github.com/aiidalab/aiidalab/pull/129 and https://github.com/aiidalab/aiidalab/pull/134 and some more cosmetic updates to the user interface.

Additional notes:
- It is easiest to test this update in combination with https://github.com/aiidalab/aiidalab/pull/133 to be able to directly modify the app registry.
- Tests are expected to fail until aforementioned PRs for aiidalab are merged.

### CODE REVIEW

**Comment by csadorf** [comment]
@yakutovicha I'll leave this PR in draft stage until the changes related to aiidalab are merged, but this is ready for testing.

**Comment by csadorf** [comment]
Merge after: https://github.com/aiidalab/aiidalab/pull/134

**Comment by yakutovicha** [comment]
@csadorf something is wrong with the tests. I tried to rerun them, and they failed.

**Comment by csadorf** [comment]
> @csadorf something is wrong with the tests. I tried to rerun them, and they failed. Can you have a look?

I don't think those tests can succeed before the aiidalab version on the docker stack is updated.

**Comment by yakutovicha** [comment]
> > @csadorf something is wrong with the tests. I tried to rerun them, and they failed. Can you have a look?
> 
> I don't think those tests can succeed before the aiidalab version on the docker stack is updated.

ok

File: home/widgets.py [hunk 1]
```diff
@@ -59,28 +61,73 @@ def _observe_message(self, change):
         self.show_temporary_message(change['new'])
 
 
-class UpdateAvailableInfoWidget(ipw.HTML):
+class AppStatusInfoWidget(ipw.HTML):
     """Widget that indicates whether an update is available."""
```
Comment (yakutovicha) [inline home/widgets.py:65]: Maybe change the docstring to better reflect the changes.

**Review by yakutovicha — COMMENTED** [review]
I think the code looks good. I also tested the behaviour and didn't notice any issues. I left one minor comment to update the docstring. Otherwise, provided that the tests are fixed - we can merge.

**Comment by csadorf** [comment]
@yakutovicha The tests still fail, because the selenium test for the app management widget fails. That's actually kind of expected, because the behavior of that widget has slightly changed (good sign that the test caught that 👍 ). That said, I've tried to re-implement the same test and had some trouble getting something proper, because I found no straight-forward way to detect whether an app is actually installed or uninstalled via selenium. It's obvious to a human, because the label texts change, but I have not yet found a way to access that information via selenium.

**Comment by yakutovicha** [comment]
> @yakutovicha The tests still fail, because the selenium test for the app management widget fails. That's actually kind of expected, because the behavior of that widget has slightly changed (good sign that the test caught that 👍 ). That said, I've tried to re-implement the same test and had some trouble getting something proper, because I found no straight-forward way to detect whether an app is actually installed or uninstalled via selenium. It's obvious to a human, because the label texts change, but I have not yet found a way to access that information via selenium.

Maybe count how many accordeons are there? Or how many app titles. They belong to a specific CSS class, so shouldn't be difficult to find how many they are:

Screenshot 2020-10-27 at 20 38 32

**Comment by csadorf** [comment]
@yakutovicha I've simplified the unit test and it now passes. It should still give us an indication if there is some issue with the app management widget.

I spent some time on trying to detect whether the button color changed (via css selection) and for some reason that only worked on my desktop. Your proposed solution is a possible option, but would be even more complicated with multiple windows etc. I'm calling it for now, because I've already spent a little too much time on this.

File: home/widgets.py (hunk 1 above)
Comment (csadorf) [reply home/widgets.py]: Done.

**Review by yakutovicha — APPROVED** [review]
Thanks, @csadorf, looks good!

**Pull request merged**